i3_workspace_names_daemon.py

```diff
@@ -135,7 +136,7 @@ def resolve_icon_or_mapping(name, leaf):
 
     def get_app_label(leaf, length):
         # interate through all identifiers, stop when first match is found
-        for identifier in ("name", "window_title", "window_instance", "window_class"):
+        for identifier in ("name", "window_title", "window_instance", "window_class", "app_id"):
```

Comment:
uh.. adding app_id ends up breaking some of i3 users configuration, because of `window_class = name`
Give me some more time to fix that

Reply:
can you give some hints what is breaking?
I have still a working i3 setup, so maybe I can help with fixing

Reply:
```
-        window_class = name
         no_match_fallback = "_no_match" in mappings and mappings["_no_match"] in fa_icons
-        if window_class:
+        if (
+            (name := getattr(leaf, 'window_class', False))
+            or (name := getattr(leaf, 'app_id', False))
+        ):
```